automation/run-tests-system.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash -ex
+
+EXEC_PATH=$(dirname "$(realpath "$0")")
+PROJECT_PATH="$(dirname $EXEC_PATH)"
+
+TEST_TYPE_ALL="all"
```

Comment:
In my opinion, there is too much duplication between this script and the container script. Maybe consider to make it one script that accepts a flag like `--test-environment bare-metal` which makes it do the preparation as done in this script. An alternative is to move the common code into a library.

Reply:
There is not that much code duplicated. They have a common structure but the container script is handling a lot of stuff related with the container (all the commands are being done with `container_exec`) and it would be quite complex to unify them.

In my opinion, they should be in different scripts to facilitate the code readability and the maintenance in the future. This script will allow more options so it will be even bigger.

What do you think? Thanks!

Reply:
More than half of the lines from `run-tests-system.sh` are copied from `run-tests.sh`, so there is quite a lot duplication in that way. You are correct that most code from `run-tests.sh` is not duplicated, though.
The container_exec function actually makes it easier to unify them because you could rename them to something generic and depending on whether it is for bare-metal or container, it would wrap the command for the container command or just execute it directly, in my opinion.